test/traits.jl

```diff
@@ -418,10 +414,10 @@ end
     @test @inferred(ccolor(Colorant{N0f8}, RGBX{Float32})) === RGBX{N0f8}
     @test @inferred(ccolor(Colorant{N0f8,3}, BGR{N0f8})) === BGR{N0f8}
 
-    @test @inferred(ccolor(AbstractRGB, HSV{Float32})) === AbstractRGB{Float32} # is not concrete type
-    @test @inferred(ccolor(AbstractRGB{N0f8}, HSV{Float32})) === AbstractRGB{N0f8} # is not concrete type
+    @test_throws TwoColorTypeError ccolor(AbstractRGB, HSV{Float32})
+    @test_throws TwoColorTypeError ccolor(AbstractRGB{N0f8}, HSV{Float32})
     @test @inferred(ccolor(AbstractRGB, RGB24)) === RGB24
-    @test_broken @inferred(ccolor(AbstractRGB{Float32}, RGB24))
+    @test_throws ErrorException ccolor(AbstractRGB{Float32}, RGB24)
```

Comment:
It's an interesting question the degree to which `ccolor` should mimic `promote_type`. Note that `promote_type` never throws an error, e.g.,
```julia
julia> promote_type(Complex{Float32}, String)
Any
```
Since `Any` is not concrete, `ccolor` cannot mimic `promote_type` exactly.

I've added some additional documentation about this.